### PR TITLE
optional and boolean api inconsistency

### PR DESCRIPTION
**Pull request by EmanueleMinotto**
This PR if for #265.

I'm not 100% sure that the documentation is enough, let me know how to edit it if it isn't good.
Thanks

### CODE REVIEW

File: readme.md [hunk 1]
```diff
@@ -292,6 +292,16 @@ $faker->optional($weight = 0.5, $default = false)->randomDigit; // 50% chance of
 $faker->optional($weight = 0.9, $default = 'abc')->word; // 10% chance of 'abc'
 ```
 
+Since v1.5 this method will accept a percentage like the `boolean()` method.
```
Comment (fzaninotto) [inline readme.md]: I think you should just replace the old doc with the new preferred way, which is the integer way. As for the BC break, it's very minimal. I'll just mention it in the release post.

File: test/Faker/Provider/BaseTest.php [hunk 2]
```diff
@@ -326,7 +326,8 @@ public function testOptionalReturnsProviderValueWhenCalledWithWeight1()
     {
         $faker = new \Faker\Generator();
         $faker->addProvider(new \Faker\Provider\Base($faker));
-        $this->assertNotNull($faker->optional(1)->randomDigit);
+        // BC: $this->assertNotNull($faker->optional(1)->randomDigit);
```
Comment (fzaninotto) [inline test/Faker/Provider/BaseTest.php]: remove it completely, commented out code is a debt. Same for the 2 other occurrences.

Comment (GrahamCampbell) [inline test/Faker/Provider/BaseTest.php]: Yeh, that's why we have git. ;)

**Comment by EmanueleMinotto** [comment]
Done, let me know if anything else is required

File: readme.md [hunk 3]
```diff
@@ -945,6 +945,7 @@ echo $faker->cityName;
 * [AliceFixturesBundle](https://github.com/h4cc/AliceFixturesBundle): A Symfony2 bundle for using Alice and Faker with data fixtures. Abled to use Doctrine ORM as well as Doctrine MongoDB ODM.
 * [Factory Muffin](https://github.com/thephpleague/factory-muffin): enable the rapid creation of objects (PHP port of factory-girl)
 * [CompanyNameGenerator](https://github.com/fzaninotto/CompanyNameGenerator): Generate names for English tech companies with class
+* [PlaceholdItProvider](https://github.com/EmanueleMinotto/PlaceholdItProvider): Generate images using placehold.it
```
Comment (fzaninotto) [inline readme.md]: Pirate change. Requires another PR ;)

**Comment by fzaninotto** [comment]
Needs rebase

File: readme.md (hunk 3 above)
Comment (EmanueleMinotto) [inline readme.md]: You're right, sorry

**Comment by EmanueleMinotto** [comment]
@fzaninotto months of delay but I hope it isn't too late

**Comment by EmanueleMinotto** [comment]
ping @fzaninotto

File: src/Faker/Provider/Base.php [hunk 4]
```diff
@@ -470,13 +470,22 @@ public static function toUpper($string = '')
...
-     * @param float $weight Set the probability of receiving a null value.
-     *                            "0" will always return null, "1" will always return the generator.
-     * @return Generator|DefaultGenerator
+     * @param float|integer $weight Set the probability of receiving a null value.
+     *                              "0" will always return null, "1" will always return the generator.
+     *                              If $weight is an integer value, then the same system works
+     *                              between 0 (always get false) and 100 (always get true).
+     * @return mixed|null
      */
     public function optional($weight = 0.5, $default = null)
```
Comment (fzaninotto) [inline src/Faker/Provider/Base.php]: shouldn't you make the default weight 50 to reflect the new system?

**Comment by fzaninotto** [comment]
It' been open for too long, time to merge!

**Pull request merged**